view/theme/frio/templates/wall_thread.tpl

```diff
@@ -81,7 +81,7 @@ as the value of $top_child_total (this is done at the end of this file)
 <span class="uriid" style="display: none;">{{$item.uriid}}</span>
 {{/if}}
 	<div class="media {{$item.shiny}}">
-	{{if $item.responses.announce && $mode != 'display'}}
+	{{if $item.reshared}}
 		<div class="wall-item-ammounce wall-item-responses" id="wall-item-ammounce-{{$item.id}}">{{$item.responses.announce.output nofilter}}</div>
 	{{/if}}
```

Comment:
This would still show "X people reshared this" as it is still using the announce item output.

Additionally, please add a prefixing reshare icon.